### DISCUSSION TITLE
configuration file option for setting the GITHUB_COM_TOKEN when targeting other platform?

### DISCUSSION THREAD

**Discussion by kastl-ars** [renovatebot/renovate#29286] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Forgejo with 37.377.4

### Please tell us more about your question or problem

I am using renovate to update things in my local Forgejo instance or in my customer's Gitlab.

I would like to set a token for GitHub to not run into API request limits.

I found that there is a warning if that token is not set:
https://docs.renovatebot.com/self-hosted-configuration/#githubtokenwarn

But it only talks about an **environment variable** `GITHUB_COM_TOKEN`. Is there a way to set this in the `config.js` configuration file?

I found only one hint in [the documentation](https://docs.renovatebot.com/self-hosted-configuration/), but that is for the [githubTokenWarn](https://docs.renovatebot.com/self-hosted-configuration/#githubtokenwarn).

Thanks in advance,
Johannes

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
Add the token in a hostrule in you conifg.js, which targets api.github.com.
eg.
```js
{
  "hostRules":[{
   matchHost: "api.github.com",
   token: "<token>"
  }],
}
```

**Reply by kastl-ars** [reply]
Thanks, that seems to work. It would be nice if this could be added to the documentation, maybe I'll find time tomorrow to open a PR.
